File: src/test/fixtures/go/math_utils.go
package main

// Add two numbers
func Add(a int, b int) int {
	return a + b
}

// Multiply two numbers
func Multiply(a int, b int) int {
	return a * b
}

// CalculateSum calculates the sum of a slice of numbers
func CalculateSum(numbers []int) int {
	total := 0
	for _, num := range numbers {
		total = Add(total, num)
	}
	return total
}

File: src/test/fixtures/python/math_utils.py
"""Math utility functions for testing LSP features."""

def add(a: int, b: int) -> int:
    """Add two numbers."""
    return a + b

def multiply(a: int, b: int) -> int:
    """Multiply two numbers."""
    return a * b

def calculate_sum(numbers: list[int]) -> int:
    """Calculate sum of a list of numbers."""
    total = 0
    for num in numbers:
        total = add(total, num)
    return total

